From 521e84fac607da89e02cdce2ef2ebf657140e4a4 Mon Sep 17 00:00:00 2001
From: Marcelo Leitner <mleitner@redhat.com>
Date: Tue, 21 Oct 2014 16:21:01 -0400
Subject: [net] netfilter: nf_tables: allow to filter from prerouting and postrouting

Message-id: <20141021162101.22888.4463.stgit@localhost.localdomain>
Patchwork-id: 98011
O-Subject: [RHEL7 net PATCH 114/124] netfilter: nf_tables: allow to filter from prerouting and postrouting
Bugzilla: 1148041 1155088
RH-Acked-by: Florian Westphal <fwestpha@redhat.com>

Backport of upstream commit: 36d2af5998258344993dd43729997a7a3baa9d99

 commit 36d2af5998258344993dd43729997a7a3baa9d99
 Author: Pablo Neira Ayuso <pablo@netfilter.org>
 Date:   Wed Oct 1 20:34:37 2014 +0200

    netfilter: nf_tables: allow to filter from prerouting and postrouting

    This allows us to emulate the NAT table in ebtables, which is actually
    a plain filter chain that hooks at prerouting, output and postrouting.

    Signed-off-by: Pablo Neira Ayuso <pablo@netfilter.org>

Signed-off-by: Marcelo Ricardo Leitner <mleitner@redhat.com>
Signed-off-by: Jiri Benc <jbenc@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/net/bridge/netfilter/nf_tables_bridge.c b/net/bridge/netfilter/nf_tables_bridge.c
index 5bcc0d8..da17a5e 100644
--- a/net/bridge/netfilter/nf_tables_bridge.c
+++ b/net/bridge/netfilter/nf_tables_bridge.c
@@ -34,9 +34,11 @@ static struct nft_af_info nft_af_bridge __read_mostly = {
  .owner  = THIS_MODULE,
  .nops  = 1,
  .hooks  = {
+  [NF_BR_PRE_ROUTING] = nft_do_chain_bridge,
   [NF_BR_LOCAL_IN] = nft_do_chain_bridge,
   [NF_BR_FORWARD]  = nft_do_chain_bridge,
   [NF_BR_LOCAL_OUT] = nft_do_chain_bridge,
+  [NF_BR_POST_ROUTING] = nft_do_chain_bridge,
  },
 };
 
-- 
1.7.1